src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -335,14 +335,32 @@ private DeclarationModifiers MakeModifiers(TypeKind typeKind, DiagnosticBag diag
             var partCount = declaration.Declarations.Length;
             var missingPartial = false;
 
+            // @davidmo On a class declaration where one has partial and the other does not, treat the other
+            // as if it did.  This is to get around adding code to classes needing partial class on any user class
+            // we want to extend as a part of our Source Generation step.
+            var hasCompilerGenAttribute = false;
+            foreach (var decl in declaration.Declarations)
+            {
+                if (decl.HasAnyAttributes)
+                {
+                    hasCompilerGenAttribute = decl.SyntaxReference.GetSyntax().ChildNodes().OfType<AttributeListSyntax>()
+                        .Where((attribList => attribList.ToString().Contains("CompilerGenerated"))).Any();
+                }
+
+                if (hasCompilerGenAttribute)
+                    break;
+            }
```

Comment:
You could make this one big LINQ statement:
```suggestion
var hasCompilerGenAttribute = declaration.Declarations
    .Any(decl =>
    {
        return decl.HasAnyAttributes &&
                decl.SyntaxReference.GetSyntax().ChildNodes().OfType<AttributeListSyntax>()
                .Where((attribList => attribList.ToString().Contains("CompilerGenerated"))).Any();
    });
```